RESULTS/FULL MAPS/map15/parameters.txt
bormann: False
cv2thresh: 150
minVal: 30
maxVal: 90
apertureSize: 5
rho: 1
theta: 0.017453292519943295
thresholdHough: 25
minLineLength: 10
maxLineGap: 5
offset: 20
diagonals: False
h: 0.023
minOffset: 1e-05
sogliaLateraleClusterMura: 5
th1: 0.1
distance_extended_segment: 20
threshold_edges: 0.1
sigma: 0.125
eps: 0.85
minPts: 1
division_threshold: 5
filter_level: 0.18
th_post: 750
th_margin: 1000
threshold_color: 200
tab_sliding: [[0.18, 0.1, 67.3211136675988, 62.7605245453973, 82.43356328563094, 78.5154934873573], [0.18, 0.1, 63.386113112891216, 59.79468707789284, 78.69382894376359, 73.20434948817058], [0.18, 0.1, 27.334850990615884, 25.520466704769518, 64.67125240422108, 58.88854235702312], [0.18, 0.1, 95.06582255749372, 92.66198887007016, 92.15365844533942, 88.06840601327708]]
tab_comparison: [[''], ['precision_micro'], ['precision_macro'], ['recall_micro'], ['recall_macro'], ['iou_micro_mean_seg_to_gt'], ['iou_macro_seg_to_gt'], ['iou_micro_mean_gt_to_seg'], ['iou_macro_gt_to_seg']]
total_index: [['map', 'freiburg_building79'], ['prec micro', 95.48956602022184], ['prec macro', 93.34769806758575], ['rec micro', 99.56691221548263], ['rec macro', 99.246511082444], ['iou mean seg', 95.06582255749372], ['iou seg', 92.66198887007016], ['iou mean gt', 92.15365844533942], ['iou gt', 88.06840601327708]]
total_index_morphological: [['map', 'freiburg_building79_1 morphological'], ['prec micro', 93.70053433247944], ['prec macro', 93.45077720207254], ['rec micro', 88.9118883992124], ['rec macro', 87.74676517728001], ['iou mean seg', 67.3211136675988], ['iou seg', 62.7605245453973], ['iou mean gt', 82.43356328563094], ['iou gt', 78.5154934873573]]
total_index_distance: [['map', 'freiburg_building79_2 distance'], ['prec micro', 96.459328532925], ['prec macro', 98.30051813471503], ['rec micro', 85.01644709571204], ['rec macro', 84.21533560064111], ['iou mean seg', 63.386113112891216], ['iou seg', 59.79468707789284], ['iou mean gt', 78.69382894376359], ['iou gt', 73.20434948817058]]
total_index_voronoi: [['map', 'freiburg_building79_3 voronoi'], ['prec micro', 96.14359412613017], ['prec macro', 98.93264248704664], ['rec micro', 64.78608243304531], ['rec macro', 59.04968531331848], ['iou mean seg', 27.334850990615884], ['iou seg', 25.520466704769518], ['iou mean gt', 64.67125240422108], ['iou gt', 58.88854235702312]]
comp: [1.5707963267948966, -1.5734121408278607, -0.0039237210494458274, 3.140284746573312]
voronoi_closeness: 10
blur: 8
iterations: 5


RESULTS/FULL MAPS/map17/parameters.txt
bormann: False
cv2thresh: 150
minVal: 30
maxVal: 90
apertureSize: 5
rho: 1
theta: 0.017453292519943295
thresholdHough: 25
minLineLength: 10
maxLineGap: 5
offset: 20
diagonals: False
h: 0.023
minOffset: 1e-05
sogliaLateraleClusterMura: 5
th1: 0.1
distance_extended_segment: 20
threshold_edges: 0.1
sigma: 0.125
eps: 0.85
minPts: 1
division_threshold: 5
filter_level: 0.18
th_post: 750
th_margin: 1000
threshold_color: 200
tab_sliding: [[0.18, 0.1, 67.3211136675988, 62.7605245453973, 82.43356328563094, 78.5154934873573], [0.18, 0.1, 63.386113112891216, 59.79468707789284, 78.69382894376359, 73.20434948817058], [0.18, 0.1, 27.334850990615884, 25.520466704769518, 64.67125240422108, 58.88854235702312], [0.18, 0.1, 95.06582255749372, 92.66198887007016, 92.15365844533942, 88.06840601327708], [0.18, 0.1, 73.17578378521368, 64.35355593532273, 68.93982632198691, 56.865550590683725], [0.18, 0.1, 65.1466068373138, 61.31036055648856, 71.97229458268006, 66.69349605356487], [0.18, 0.1, 20.676385950949445, 22.99050902164426, 28.488895651989708, 17.48166953734271], [0.18, 0.1, 90.26679259585255, 91.92735570980317, 88.72932535602752, 84.52059290050867], [0.18, 0.1, 55.64247556968491, 47.68716045750032, 38.84060129673714, 21.170227041810712], [0.18, 0.1, 52.387143127376426, 74.8781048380477, 45.059148499495336, 21.593030202301726], [0.18, 0.1, 44.22804287870198, 15.07433078201926, 72.75413411777821, 45.29714390173838], [0.18, 0.1, 80.44631877045927, 56.23209128728812, 80.29568283374573, 48.66443737158052]]
tab_comparison: [[''], ['precision_micro'], ['precision_macro'], ['recall_micro'], ['recall_macro'], ['iou_micro_mean_seg_to_gt'], ['iou_macro_seg_to_gt'], ['iou_micro_mean_gt_to_seg'], ['iou_macro_gt_to_seg']]
total_index: [['map', 'freiburg_building79', 'map15', 'freiburg_building101'], ['prec micro', 95.48956602022184, 95.211427163364, 99.07135875802075], ['prec macro', 93.34769806758575, 95.74943517687475, 98.80022155842614], ['rec micro', 99.56691221548263, 97.05633736827721, 90.84612992504174], ['rec macro', 99.246511082444, 97.41289643390719, 71.85746296773682], ['iou mean seg', 95.06582255749372, 90.26679259585255, 80.44631877045927], ['iou seg', 92.66198887007016, 91.92735570980317, 56.23209128728812], ['iou mean gt', 92.15365844533942, 88.72932535602752, 80.29568283374573], ['iou gt', 88.06840601327708, 84.52059290050867, 48.66443737158052]]
total_index_morphological: [['map', 'freiburg_building79_1 morphological', 'map15_1 morphological', 'freiburg_building101_1 morphological'], ['prec micro', 93.70053433247944, 80.76100817737701, 80.95388062008732], ['prec macro', 93.45077720207254, 72.72384017826747, 87.65876036700189], ['rec micro', 88.9118883992124, 92.77042089741813, 88.591682811582], ['rec macro', 87.74676517728001, 93.70181488088859, 65.69533403398839], ['iou mean seg', 67.3211136675988, 73.17578378521368, 55.64247556968491], ['iou seg', 62.7605245453973, 64.35355593532273, 47.68716045750032], ['iou mean gt', 82.43356328563094, 68.93982632198691, 38.84060129673714], ['iou gt', 78.5154934873573, 56.865550590683725, 21.170227041810712]]
total_index_distance: [['map', 'freiburg_building79_2 distance', 'map15_2 distance', 'freiburg_building101_2 distance'], ['prec micro', 96.459328532925, 90.3596084214838, 90.17705319037492], ['prec macro', 98.30051813471503, 91.51867784964872, 91.37084819103329], ['rec micro', 85.01644709571204, 81.90217440206999, 84.27090840776961], ['rec macro', 84.21533560064111, 83.57529643895029, 92.10275833349858], ['iou mean seg', 63.386113112891216, 65.1466068373138, 52.387143127376426], ['iou seg', 59.79468707789284, 61.31036055648856, 74.8781048380477], ['iou mean gt', 78.69382894376359, 71.97229458268006, 45.059148499495336], ['iou gt', 73.20434948817058, 66.69349605356487, 21.593030202301726]]
total_index_voronoi: [['map', 'freiburg_building79_3 voronoi', 'map15_3 voronoi', 'freiburg_building101_3 voronoi'], ['prec micro', 96.14359412613017, 72.96731599862233, 90.99480950329252], ['prec macro', 98.93264248704664, 51.84921543746711, 94.00016026283105], ['rec micro', 64.78608243304531, 81.22567717427381, 82.46793801393592], ['rec macro', 59.04968531331848, 73.70660581341839, 51.59332923516231], ['iou mean seg', 27.334850990615884, 20.676385950949445, 44.22804287870198], ['iou seg', 25.520466704769518, 22.99050902164426, 15.07433078201926], ['iou mean gt', 64.67125240422108, 28.488895651989708, 72.75413411777821], ['iou gt', 58.88854235702312, 17.48166953734271, 45.29714390173838]]
comp: [0.013425609630725699, -3.1261015655543405, -1.5645998915807153, 1.5749272836043513]
voronoi_closeness: 10
blur: 8
iterations: 5


code/parameters.py
import numpy as np

# -------------------------------
# faces ("celle" ) classification_method choice (2 is older version)
metodo_classificazione_celle = 1
# metodo_classificazione_celle = 2
# -------------------------------


class PathObj:
    def __init__(self):
        self.name_folder_input = ''
        self.path_folder_input = ''
        self.path_folder_output = ''
        self.path_log_folder = ''
        # name of image
        self.metric_map_name = ''
        self.metric_map_path = ''
        self.path_xml = ''
        self.filepath = ''
        self.path_orebro = ''
        self.orebro_img = ''
        self.gt_color = ''


class ParameterObj:
    def __init__(self):
        # bormann
        self.bormann = False

        # Thresholding parameters
        self.cv2thresh = 150

        # TODO  we don't use canny anymore. so we can delete this parameters. .
        # Canny parameters
        self.minVal = 30
        self.maxVal = 90
        self.apertureSize = 5

        # Hough parameters
        self.rho = 1
        self.theta = np.pi / 180
        self.thresholdHough = 25
        self.minLineLength = 10
        self.maxLineGap = 5

        # Offset for x_min,x_max,y_min,y_max
        self.offset = 20

        # Angular Clusters parameters, mean-shift parameters
        # diagonals. if there is some obliques lines set diagonals=False
        self.diagonals = False
        self.h = 0.023
        self.minOffset = 0.00001

        # Spatial cluster parameters
        self.sogliaLateraleClusterMura = 5

        # Extended lines parameters
        self.th1 = 0.1
        self.distance_extended_segment = 20

        # Edges parameters
        self.threshold_edges = 0.1

        # Matrix parameters
        # parameter to check the weight of an edge for clustering of faces
        self.sigma = 0.125

        # DBSCAN parameters
        self.eps = 0.85  # 0.85#1.5#0.85
        self.minPts = 1

        # Cells classification parameters
        self.division_threshold = 5

        # Orebro parameters
        self.filter_level = 0.18

        # post processing parameters
        self.th_post = 750
        # TODO maybe this parameter is not needed anymore (th_margin)
        self.th_margin = 1000

        # evaluation parameters
        self.threshold_color = 200
        self.tab_sliding = []
        self.tab_comparison = None
        # self.total_index = [['map', 'prec micro', 'prec macro', 'rec micro', 'rec macro', 'iou mean seg', 'iou seg',
        #                     'iou mean gt', 'iou gt'], [[], [], [], [], [], [], [], [], []]]
        self.total_index = [['map'], ['prec micro'], ['prec macro'], ['rec micro'], ['rec macro'], ['iou mean seg'],
                            ['iou seg'], ['iou mean gt'], ['iou gt']]
        self.total_index_morphological = [['map'], ['prec micro'], ['prec macro'], ['rec micro'], ['rec macro'], ['iou mean seg'],
                            ['iou seg'], ['iou mean gt'], ['iou gt']]
        self.total_index_distance = [['map'], ['prec micro'], ['prec macro'], ['rec micro'], ['rec macro'], ['iou mean seg'],
                            ['iou seg'], ['iou mean gt'], ['iou gt']]
        self.total_index_voronoi = [['map'], ['prec micro'], ['prec macro'], ['rec micro'], ['rec macro'], ['iou mean seg'],
                            ['iou seg'], ['iou mean gt'], ['iou gt']]

        # integration from orebro
        self.comp = None

        # voronoi parameters
        self.voronoi_closeness = 10
        self.blur = 8
        self.iterations = 5

    def set_threshold_hough(self, value):
        self.thresholdHough = value

    def set_sigma(self, value):
        self.sigma = value

    def set_eps(self, value):
        self.eps = value

    def set_filter_level(self, value):
        self.filter_level = value

    def set_th1(self, value):
        self.th1 = value

    def set_distance_extended_segment(self, value):
        self.distance_extended_segment = value


class ParameterDraw:
    def __init__(self):
        self.map = True
        self.canny = False
        self.hough = True
        self.walls = True
        self.contour = False
        self.angular_cluster = True
        self.representative_segments = True
        self.spatial_wall_cluster = True
        self.spatial_cluster = True
        self.extended_lines = True
        self.edges = True
        self.dbscan = True
        self.cells_in_out = True
        self.rooms = True
        self.rooms_on_map = True
        self.rooms_on_map_prediction = True
        self.rooms_on_map_lines = True
        self.sides = False
        self.complete = True
